### PR TITLE
Merge/main to pbft 0.50.0

### PR DESCRIPTION
**Pull request by limebell**
Remarkable conflics:
- `BlockChain.cs`

### CODE REVIEW

**Comment by riemannulus** [comment]
Could you check CI one more time?

File: Libplanet/Blockchain/BlockChain.cs [hunk 1]
```diff
@@ -1784,5 +1760,71 @@ public InvalidBlockException ValidateNextBlock(Block<T> block)
...
+
+            if (block.PreviousHash is { } previousHash)
+            {
+                return new InvalidBlockPreviousHashException(
+                    $"A genesis block should not have previous hash, " +
+                    $"but its value is {block.PreviousHash}.");
+            }
+
+            return null;
+        }
```
Comment (greymistcube) [inline Libplanet/Blockchain/BlockChain.cs:1800]: We probably need `LastCommit` validation somewhere in here. 🙃

Comment (limebell) [reply Libplanet/Blockchain/BlockChain.cs:1800]: You mean `GenesisBlock` should have `null` `LastCommit`?

Comment (greymistcube) [reply Libplanet/Blockchain/BlockChain.cs:1800]: Yeah, this bypasses the rest of `ValidateNextBlock()` so it won't be checked.

Comment (limebell) [reply Libplanet/Blockchain/BlockChain.cs]: Great, added in https://github.com/planetarium/libplanet/pull/2871/commits/65e1e13cad23d7fe9c1b98b03bf2d6a538dd2f4b.

**Review by greymistcube — APPROVED** [review]

**Pull request merged**